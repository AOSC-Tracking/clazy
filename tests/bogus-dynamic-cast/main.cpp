















struct A
{

};

struct B : public A
{

};

void test()
{
    A *a = new A();
    B *b = new B();

    dynamic_cast<A*>(b); // warning: Casting to base
    dynamic_cast<A*>(a); // warning: Casting to itself
    dynamic_cast<B*>(a); // OK
    dynamic_cast<B*>(b); // warning: Casting to itself
}
