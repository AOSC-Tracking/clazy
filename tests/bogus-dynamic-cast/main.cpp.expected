main.cpp:32:5: warning: explicitly casting to base is unnecessary [-Wclazy-bogus-dynamic-cast]
    dynamic_cast<A*>(b); // warning: Casting to base
main.cpp:33:5: warning: Casting to itself [-Wclazy-bogus-dynamic-cast]
    dynamic_cast<A*>(a); // warning: Casting to itself
main.cpp:35:5: warning: Casting to itself [-Wclazy-bogus-dynamic-cast]
    dynamic_cast<B*>(b); // warning: Casting to itself
